### ISSUE TITLE
Excel file not correct. (Can not be opened)  with decimal columns.             Sample5.xlsx Do not works

### ISSUE THREAD

**Issue by melDavila** [mbleron/ExcelGen#11]
If I try to create dummy Excel BLOB with natural number columns (1, 2, 3, 20, 50) it works.
But with columns with decimals the excel file is not correct (3.1415).

  FUNCTION get_EXCEL_FROM_QUERY  RETURN BLOB as 
    v_blob BLOB ;
    v_ctxId         ExcelGen.ctxHandle; 
    BEGIN
      v_ctxId := ExcelGen.createContext();
      ExcelGen.addSheetFromQuery (v_ctxId, 'TEST','SELECT  1 n1,2 n2 , **3.1415** pi from dual');
      ExcelGen.setNumFormat       (v_ctxId, '0.0000'  );
      v_blob := ExcelGen.getFileContent(v_ctxId);
      ExcelGen.closeContext(v_ctxId);
      return v_blob;
  END get_EXCEL_FROM_QUERY;

Change **3.1415** to **31415** and it will work. 

To test from SQL DEveloper:
SELECT EXCEL_GEN.get_EXCEL_FROM_QUERY   from dual; 

The example : 
**_Setting sheet or column-level cell formats : sample5.xlsx_**
DO NOT WORKS

**Comment by mbleron** [comment]
I don't reproduce this one.
What's your database version?
Attach the output file here please.

**Comment by melDavila** [comment]
It was a setting problem.
Do not happen doing 

### alter session set NLS_NUMERIC_CHARACTERS='.,';

**Comment by mbleron** [comment]
It's a bug actually. The program is not meant to rely on NLS settings.
This is the culprit : 
```
if sd.sqlMetadata.columnList(i).scale <= 0 then
  data.varchar2_value := to_char(data.number_value);
else
  data.varchar2_value := to_char(data.number_value, 'TM9', NLS_PARAM_STRING);
end if;
```
For a NUMBER value, without any precision or scale defined (such as a number literal in the example), Oracle reports the scale as -127, and precision as 0.
Therefore, the number-to-string conversion is performed using a simple TO_CHAR, without format, and inherits the session NLS.

I'll fix that tomorrow.

**Comment by mbleron** [comment]
Fixed in v2.2.1
